code/game/machinery/_machines_base/stock_parts/_stock_parts.dm

```diff
@@ -86,18 +89,19 @@
 	SetName("broken [name]")
 
 /obj/item/stock_parts/proc/is_functional()
-	return health > 0
+	return (health == -1) || (health > 0)
```

Comment:
Should probably have a macro for `-1` to make it a bit more descriptive.

Reply:
That's probably true. But, I'm afraid having an HEALTH_NO_DAMAGE or something among the line as a constant would cause confusion with the other existing health systems that may or may not handle -1 as being invulnerable. Also, -1 is usually used as an invalid/nonsensical value that's not null. 

I'd say that might be best reserved for whoever reworks the health system as a whole. I just needed items to handle health properly to make life easier, so I'd rather not get into the massive headache that unifying all those would be. At least not for now XD

Reply:
A given constant meaning something else in another system isn't a good reason not to make it clear in this system. I am not in favour of leaving a bunch of bare-assed `-1`s in this code.

Reply:
Alright then